gb-chris/scripts/article.js

```diff
@@ -0,0 +1,68 @@
+Article.all = [];
+
+// COMMENT: Why isn't this method written as an arrow function?
+// The method uses a contextualized this that requires to be scoped towards the Article instances, making it an arrow function would scope the this to be on window at this level.
+Article.prototype.toHtml = function() {
+  let template = Handlebars.compile($('#article-template').text());
+
+  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+
+  // COMMENT: What is going on in the line below? What do the question mark and colon represent? How have we seen this same logic represented previously?
+  // Not sure? Check the docs!
+  // This is a turnary operator that functions as a conditional. The statement is saying IF this(Article instance).publishedOn is true evaluate to the template literal, otherwise the published status is draft.
+  this.publishStatus = this.publishedOn ? `published ${this.daysAgo} days ago` : '(draft)';
+  this.body = marked(this.body);
+
+  return template(this);
+};
+
+// REVIEW: There are some other functions that also relate to all articles across the board, rather than just single instances. Object-oriented programming would call these "class-level" functions, that are relevant to the entire "class" of objects that are Articles.
+
+// REVIEW: This function will take the rawData, how ever it is provided, and use it to instantiate all the articles. This code is moved from elsewhere, and encapsulated in a simply-named function for clarity.
+
+// COMMENT: Where is this function called? What does 'rawData' represent now? How is this different from previous labs?
+// The rawData here is now in local storage. Unlike prior labs where our raw data was a local .js file that we drew our articles from. It is now a local storage item that we draw out or load from a server.
+Article.loadAll = articleData => {
+  articleData.sort((a,b) => (new Date(b.publishedOn)) - (new Date(a.publishedOn)))
+
+  articleData.forEach(articleObject => Article.all.push(new Article(articleObject)))
+}
+
+// REVIEW: This function will retrieve the data from either a local or remote source, and process it, then hand off control to the View.
+Article.fetchAll = () => {
+  // REVIEW: What is this 'if' statement checking for? Where was the rawData set to local storage?
+  if (localStorage.rawData) {
+
+    // Article.loadAll(localStorage.getItem('rawData', JSON.parse(rawData)));
+    Article.loadAll(JSON.parse(localStorage.getItem('rawData')));
+    articleView.initIndexPage();
+
+  // If local storage exist load local data an init the page, if it doesnt exist load jquery ajax to draw data with get method. on sucess of that draw stringify and set info under raw data, load articles and draw data. Initialized rae data on index page. If it is not a successful get and error will occur.
```

Comment:
Great description of what's happening here! Instead of "on success of that draw", I'd call it success of the fetch, or success of the AJAX call.